network/memsocket/src/lib.rs

```diff
@@ -90,8 +91,11 @@ impl MemoryListener {
     /// ```
     ///
     /// [`local_addr`]: #method.local_addr
-    pub fn bind(port: u16) -> Result<Self> {
-        let mut switchboard = (&*SWITCHBOARD).lock().unwrap();
+    pub fn bind(port: u16) -> Result<Self,> {
+        let mut switchboard =  match (&*SWITCHBOARD).lock() {
+            Ok(switchboard) => switchboard,
+            Err(_e) => return Err(ErrorKind::AddrInUse.into()),
+        };
```

Comment:
Have we seen this be a problem in fuzzing or something?

`MemSocket` isn't used in the real world because we use `TcpSocket`s